lib/logstash/string_interpolation.rb

```diff
@@ -0,0 +1,155 @@
+require "thread_safe"
+require "forwardable"
+
+module LogStash
+  class StringInterpolation
+    # Floats outside of these upper and lower bounds are forcibly converted
+    # to scientific notation by Float#to_s
+    MIN_FLOAT_BEFORE_SCI_NOT = 0.0001
+    MAX_FLOAT_BEFORE_SCI_NOT = 1000000000000000.0
+
+    INSTANCE = StringInterpolation.new
+    CACHE = ThreadSafe::Cache.new
+    TEMPLATE_TAG_REGEXP = /%\{[^}]+\}/
+
+    def evaluate(event, template)
```

Comment:
@colinsurprenant I think this should be protected since it's a singleton ,what do you think?